src/repr/src/scalar.rs

```diff
@@ -1448,6 +1466,29 @@ impl<'a> ScalarType {
                 value_type: Box::new(value_type.default_embedded_value()),
                 custom_oid: None,
             },
+            Record {
+                fields,
+                custom_oid,
+                custom_name,
+            } => {
+                let default_embedded_field_values = fields
+                    .iter()
+                    .map(|(column_name, column_type)| {
+                        (
+                            column_name.clone(),
+                            ColumnType {
+                                scalar_type: column_type.scalar_type.default_embedded_value(),
+                                nullable: column_type.nullable,
+                            },
+                        )
+                    })
+                    .collect_vec();
+                Record {
+                    fields: default_embedded_field_values,
+                    custom_name: custom_name.clone(),
```

Comment:
I will need to think about this, but there's a chance that this violates the spirit of this function, in that it propagates the name/oid of the type, rather than its..."desaturated" value, which this function's meant to return.

Reply:
I'm almost wondering if we don't want a `Record` match at all, and fall through to the default `v.clone()`. That's what happens for existing List/Map types with custom oids already.

If we don't propagate the name/oid at all, then `pg_typeof` will return a generic Row rather than the custom implementation, which breaks from postgres behavior